Fonctionnement du worker - idle - consommation de RAM

La consommation de RAM de notre serveur de BDD semble anormalement élevée et lorsque l'on redémarre GeoNature, cette consommation s'effondre avant de remonter progressivement par plateau (la BDD est sur un serveur dédié, dissocié de nos instances GN).
Le problème rappelle https://github.com/PnX-SI/GeoNature/issues/2717
En investiguant, on trouve régulièrement des sessions dont le statut est `idle`.
On a testé de paramétrer `idle_session_timeout` à 15 minutes pour voir l'effet.
Ce paramétrage ayant moins de 24 heures il est dur de conclure sur l'impact par contre, il semble avoir eu un impact sur les taches du backend dont certaines ont planté, par exemple :

`[2025-04-09 01:00:00,101: ERROR/ForkPoolWorker-8] Task geonature.core.gn_commons.tasks.clean_attachments[8f006f02-94f9-4229-90b8-33bacf40e570] raised unexpected: OperationalError('(psycopg2.errors.IdleSessionTimeout) terminating connection due to idle-session timeout\nSSL connection has been closed unexpectedly\n')
`
Est-ce que ce comportement de session `idle` est normal ? Est-ce certains ont testé d'autres paramètres ?
De notre côté, on va essayer de l'augmenter mais je suis preneur de vos retours / avis.
Merci

J'ai donné celle-ci en exemple mais il y en a d'autres (export par exemple) ou celle-ci qui revient plusieurs fois : 

```
[2025-04-09 04:31:09 +0000] [8] [ERROR] Exception on /gn_commons/list/parameters [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.IdleSessionTimeout: arr��t des connexions suite �� l'expiration du d��lai d'inactivit�� de la session
SSL connection has been closed unexpectedly

```
Ce que je ne comprends pas, c'est pourquoi ces sessions ne sont pas fermées à la fin de l'exécution des tâches.
On va laisser ce paramétrage comme ça aujourd'hui et l'augmenter en de journée pour voir si ça limite les erreurs.

--- a/backend/geonature/utils/config_schema.py
+++ b/backend/geonature/utils/config_schema.py
@@ -197,6 +197,8 @@ class GnPySchemaConf(Schema):
         ),
     )
     SQLALCHEMY_TRACK_MODIFICATIONS = fields.Boolean(load_default=True)
+
+    SQLALCHEMY_ENGINE_OPTIONS = fields.Dict(load_default={})
     SESSION_TYPE = fields.String(load_default="filesystem")
     SECRET_KEY = fields.String(required=True, validate=Length(min=20))
     # le cookie expire toute les 7 jours par défaut

--- a/config/default_config.toml.example
+++ b/config/default_config.toml.example
@@ -20,6 +20,11 @@ SQLALCHEMY_DATABASE_URI = "postgresql://monuser:monpassachanger@localhost:monpor
 # Note : Patch pour suivre la supression des médias et pouvoir supprimer un fichier sur un DELETE de media
 SQLALCHEMY_TRACK_MODIFICATIONS = true
 
+# Ce paramètre permet de configurer la connexion de SQLAlchemy avec la base de données
+# Les paramètres de configuration de la connexion doivent correspondre à ceux utilisé dans la fonction `create_engine` 
+# de SQLAlchemy : https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine
+SQLALCHEMY_ENGINE_OPTIONS = {}
+
 # Application
 
 # Nom de l'application dans la page d'accueil.

